### PR TITLE
Add demo for lecture 3 - viewmodels.

### PR DESCRIPTION
**Pull request by simona-kurnavova**
What was added:
- ProfileViewModel for ProfileScreen (copy from last year with changed LiveData to StateFlow) and usage
- Small UI changes on Home screen: input field for username search and title
- Added memory repository (copy from last year)
- Loading of image in ProfileScreen using Coil

### CODE REVIEW

File: Lecture 3/GithubApp/app/src/main/java/mff/mdp/demoapp/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        super.onCreate(savedInstanceState)
+        enableEdgeToEdge()
+        setContent {
+            DemoAppTheme {
+                Scaffold(
+                    modifier = Modifier.fillMaxSize(),
+                    topBar = {
+                        TopAppBar(title = { Text(text = getString(R.string.app_name)) })
+                    }
+                ) { innerPadding ->
```
Comment (sakulisko) [inline Lecture 3/GithubApp/app/src/main/java/mff/mdp/demoapp/MainActivity.kt]: For educational purposed, it is maybe better to use `content` as named parameter

File: Lecture 3/GithubApp/app/src/main/java/mff/mdp/demoapp/ui/UserCard.kt [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import coil.compose.AsyncImage
+import mff.mdp.demoapp.R
+
+@Composable
+fun UserCard(
+    name: String,
+    url: String,
+    imageUrl: String,
+    modifier: Modifier = Modifier
+) {
```
Comment (sakulisko) [inline Lecture 3/GithubApp/app/src/main/java/mff/mdp/demoapp/ui/UserCard.kt:35]: This is maybe just matter of style, but I would provide also Composable `fun UserCard(user: User, modifier: Modifier = Modifier)` which will call this function

File: Lecture 3/GithubApp/app/src/main/java/mff/mdp/demoapp/ui/theme/Theme.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+private val LightColorScheme = lightColorScheme(
+    primary = Purple40,
+    secondary = PurpleGrey40,
+    tertiary = Pink40,
+    onPrimary = Color.White,
+    onSecondary = Color.White
+)
+
+@Composable
+fun DemoAppTheme(
```
Comment (sakulisko) [inline Lecture 3/GithubApp/app/src/main/java/mff/mdp/demoapp/ui/theme/Theme.kt]: Maybe `GithubAppTheme` is better fit?

File: Lesson 2/DemoApp/app/src/main/java/mff/mdp/demoapp/data/GithubRepository.kt [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.gendigital.mff.lecture3.data
```
Comment (sakulisko) [inline Lesson 2/DemoApp/app/src/main/java/mff/mdp/demoapp/data/GithubRepository.kt]: Copy&Paste error, please fix the package names, I prefer to use the "com.gendigital....` package. Please check also other classes

File: Lecture 3/GithubApp/app/src/main/java/mff/mdp/demoapp/ui/UserCard.kt (hunk 2 above)
Comment (simona-kurnavova) [reply Lecture 3/GithubApp/app/src/main/java/mff/mdp/demoapp/ui/UserCard.kt:35]: Sure

File: Lesson 2/DemoApp/app/src/main/java/mff/mdp/demoapp/data/GithubRepository.kt (hunk 4 above)
Comment (simona-kurnavova) [reply Lesson 2/DemoApp/app/src/main/java/mff/mdp/demoapp/data/GithubRepository.kt]: It's worse 😆 I pushed completely different project as well

**Pull request merged**